MonoDevelop.FSharpBinding/FakeSearchCategory.fs

```diff
@@ -0,0 +1,160 @@
+          
+          let shouldMatch = PropertyService.Get ("FSharpBinding.MatchWithThemePropName", false)
+          let themeTextColour = colourStyles.PlainText.Foreground |> cairoToGdk
+          let themeBackColour = colourStyles.PlainText.Background |> cairoToGdk
+
+          if shouldMatch then
+              v.ModifyText(Gtk.StateType.Normal, themeTextColour)
+              v.ModifyBase(Gtk.StateType.Normal, themeBackColour)
+          else
+              let textColour = PropertyService.Get ("FSharpBinding.TextColorPropName", "#000000") |> ColorHelpers.strToColor
+              let backColour = PropertyService.Get ("FSharpBinding.BaseColorPropName", "#FFFFFF") |> ColorHelpers.strToColor
+              v.ModifyText(Gtk.StateType.Normal, textColour)
+              v.ModifyBase(Gtk.StateType.Normal, backColour)
+    | _ -> ()
+    
+  member x.UpdateFont() = 
+    let fontName = MonoDevelop.Ide.Fonts.FontService.MonospaceFont.Family
+    let fontName = PropertyService.Get ("FSharpBinding.FsiFontName", fontName)
+    LoggingService.LogDebug ("FAKE task runner: Loading font '{0}'", fontName)
+    let font = Pango.FontDescription.FromString(fontName)
+    view.SetFont(font)
+
+type FakeSearchResult(solution: Solution, match', matchedString, rank, scriptPath) =
+  inherit SearchResult(match', matchedString, rank)
+
+  override x.SearchResultType =
+    SearchResultType.Type
+
+  override x.Description =
+    sprintf "Runs the FAKE %s task in %s" matchedString solution.Name
+
+  override x.PlainText = "FAKE " + matchedString
+
+  override x.Icon =
+    getImage "md-command"
+
+  override x.CanActivate = true
+
+  override x.Activate() =
+    let pad = IdeApp.Workbench.GetPad<FakePad>()
+
+    pad.BringToFront()
+    let padContent = pad.Content :?> FakePad
+
+    padContent.Run (string solution.BaseDirectory, matchedString, scriptPath)
+
+type FakeSearchCategory() =
+  inherit SearchCategory("FAKE", sortOrder = SearchCategory.FirstCategory)
+
+  override x.get_Tags() = [|"fake"|]
+
+  override x.IsValidTag _tag =
+    true
+
+  override x.GetResults(searchCallback, pattern, token) =
+
+    let addResult (solution: Solution, m: Match, rank, scriptPath) = 
+      if token.IsCancellationRequested then ()
+      else
+        let sr = FakeSearchResult(solution, pattern.Pattern, m.Groups.[1].Value, rank, scriptPath)
+        searchCallback.ReportResult sr
+
+    Task.Run(
+      (fun () ->
+         let matcher = StringMatcher.GetMatcher (pattern.Pattern, false)
+         async {
+           for solution in IdeApp.Workspace.GetAllSolutions() do
+
+             let launcherScript = if Platform.IsWindows then
+                                    "build.cmd"
+                                  else
+                                    "build.sh"
+
+             let fakeScriptPath = Path.Combine([|string solution.BaseDirectory; "build.fsx"|])
```

Comment:
FAKE works with non F#, so I guess so. FAKE appears to work at the solution level rather than the project level.